airflow/bin/cli.py

```diff
@@ -424,15 +427,72 @@ def clear(args):
         confirm_prompt=not args.no_confirm,
         include_subdags=not args.exclude_subdags)
 
+def command_of(pid):
+    """
+    Returns the command (ie argv[0]) of the process with process id @pid
+    """
+    return subprocess.check_output([
+        'ps', str(pid), '-o', 'command'
+    ]).split('\n')[1]
+
+def restart_workers(gunicorn_master_proc, num_workers_expected):
+    """
+    Runs forever, monitoring the child processes of @gunicorn_master_proc and restarting workers
+    occasionally.
+
+    State transition diagram, where each state is [$NUM_READY_WORKER_PROCS_FOUND / $NUM_WORKER_PROCS_FOUND].
+    We expect most time to be spent in [n / n]. The horizontal transition at ? happens when the new worker
+    parses all the dags.
+
+       V ------------------------------------------------------------------------------------------------------|
+    [n / n] ----$WORKER_REFRESH_WAIT----> [n / n + 1]  ----$WORKER_STARTUP_WAIT----?----> [n + 1 / n + 1] ---- |
+                                               ^-----------------------------------|
+
+    We change the number of workers by sending TTIN and TTOU to the gunicorn master process, which increases
+    and decreases the number of child workers respectively. Gunicorn guarantees that on TTOU workers are terminated
+    gracefully and that the oldest worker is terminated.
+    """
+
+    while True:
+
+        workers = psutil.Process(gunicorn_master_proc.pid).children()
+        ready_workers = [
+            proc for proc in workers if '[ready]' in command_of(proc.pid)
```

Comment:
We're setting the command string ourselves at https://github.com/apache/incubator-airflow/pull/1685/files#diff-e4871ae5b73f6d64aff5bfba50a00c24R19

As for alternatives, we can consider other approaches like

- Writing the worker state into a file
- Shared memory
- python Pipes / Queues

but I think changing the command string is the simplest and gives the most visibility